src/sharing/components/ShareAutocomplete.jsx

```diff
@@ -23,39 +23,70 @@ export default class ShareAutocomplete extends Component {
     )
   }
 
-  onSuggestionsFetchRequested ({ value }) {
-    this.setState(state => ({
-      ...state,
-      suggestions: this.computeSuggestions(value)
-    }))
+  onSuggestionsFetchRequested = ({ value }) => {
+    this.setState(state => ({...state, suggestions: this.computeSuggestions(value)}))
+  }
+
+  onSuggestionsClearRequested = () => {
+    this.setState(state => ({...state, suggestions: []}))
+  }
+
+  onChange = (event, { newValue, method }) => {
+    // A suggestion has been picked
+    if (typeof newValue === 'object') {
+      this.onPick(newValue)
+    } else {
+      this.setState(state => ({...state, inputValue: newValue}))
+    }
+  }
+
+  onKeyPress = (event) => {
+    // The user wants to add an unknown email
+    if (event.key === 'Enter' && this.state.inputValue !== '') {
+      this.onPick({ email: this.state.inputValue })
+    }
   }
 
-  onSuggestionsClearRequested () {
-    this.setState(state => ({
-      ...state,
-      suggestions: []
-    }))
+  onPick = (value) => {
+    this.props.onPick(value)
+    this.setState(state => ({...state, inputValue: ''}))
+    setTimeout(() => this.input.focus(), 1) // don't ask...
   }
 
-  onChange (event, { newValue }) {
-    const email = getPrimaryEmail(newValue) || newValue
-    const url = getPrimaryCozy(newValue)
-    const id = newValue._id
-    this.props.onChange(email, url, id)
+  onRemove = (value) => {
+    this.props.onRemove(value)
+    setTimeout(() => this.input.focus(), 1)
```

Comment:
Yet another `don't ask...` mystery. What's that?

Reply:
I said: don't ask! :p

Reply:
I know, that's exactly why I asked!
;-)